groupmne/group_model.py

```diff
@@ -120,36 +120,52 @@ def _group_filtering(fwds, src_ref, noise_covs=None):
         subject = src[0]["subject_his_id"]
         group_info["subjects"].append(subject)
         ch_names.append(utils._get_channels(fwd, cov))
-        # find removed vertices
+
+        # find mapping between src_ref and new src space of fwd
         mapping = utils.get_morph_src_mapping(src_ref, src)
-        gain = []
+
+        # create a gain with the "large" src_ref resolution
+        gain = np.ones((fwd["nchan"], sum(n_sources)))
         for i in range(2):
+            # pos contains the reference sources of src_ref
             pos = list(mapping[0][i].keys())
             positions[i].append(pos)
             vertno = - np.ones(n_sources[i]).astype(int)
-            gain_h = np.ones((fwd["nchan"], n_sources[i]))
             # re-order columns of the gain matrices
+            # vertno_tmp contains the new vertices in the right order
             vertno_tmp = np.array(list(mapping[0][i].values()))
+            # find the appropriate permutation to reorder columns
             permutation = np.argsort(np.argsort(vertno_tmp))
-            gain_h[:, pos] = fwd["sol"]["data"][:, permutation]
+
+            # these indices allow to switch between left and right hemispheres
+            col_0 = i * n_sources[i]
```

Comment:
this seems suspicious. Usually when you loop over sources spaces / hemispheres you sum the number of locations you have seen so far.